doc/overview.md

```diff
@@ -0,0 +1,124 @@
+
+## Welcome to libstorage-ng
+
+
+libstorage-ng is the designated successor of
+[libstorage](https://github.com/openSUSE/libstorage), a C++ library used by
+[YaST](https://github.com/yast) to perform most storage related tasks.
+
+The main idea of libstorage-ng compared to libstorage is to keep all
+information about storage devices in a
+[graph](https://en.wikipedia.org/wiki/Graph_(discrete_mathematics)) instead of
+lists. A second and even more interesting use-case for a graph are the actions
+performed when committing the queued changes to the system.
+
+
+### Device Graph
+
+Storing the devices in libstorage as a graph looks like a natural
+approach. Here is a simple graph showing a disk with a partition table, two
+partitions and two filesystems where one is encrypted:
+
+![luks.png](images/luks.png)
+
+You might have seen the diagram above in YaST. But in libstorage-ng it is a
+direct dump of internal data structure instead of a [cumbersome
+transformation](https://github.com/openSUSE/libstorage/blob/master/storage/Graph.cc).
+
+The graph approach has several advantages compared to the list of list design
+of libstorage:
+
+* No pseudo containers for e.g. MD RAID, tmpfs or NFS are needed.
+
+* Possibility to have special objects for IMSM and DDF RAID containers.
+
+* Some operations are trivial compared to the current libstorage, e.g. getting
+  all devices using a device is just a plain
+  [BFS](https://en.wikipedia.org/wiki/Breadth-first_search).
+
+Together with a redesign of the device objects several features will be
+simple, e.g. using unpartitioned disks for filesystems, renaming LVM volume
```

Comment:
Just a couple of minor fixes for this paragraph that, at least to me, could make it more readable:

The device objects has also being redesigned and several features are now simple to implement and use, e.g. using unpartitioned disks for filesystems, renaming LVM volume groups and logical volumes or switching between partitioned and unpartitioned MD RAID.